dbms/src/Flash/Executor/DataStreamExecutor.cpp

```diff
@@ -65,11 +69,23 @@ String DataStreamExecutor::toString() const
 
 int DataStreamExecutor::estimateNewThreadCount()
 {
-    return data_stream->estimateNewThreadCount();
+    return estimate_thread_cnt;
 }
 
 RU DataStreamExecutor::collectRequestUnit()
 {
-    return toRU(data_stream->estimateCPUTimeNs());
+    auto origin_cpu_time_ns = data_stream->estimateCPUTimeNs();
+    UInt64 total_thread_cnt = thread_cnt_before_execute + estimate_thread_cnt;
+    size_t physical_cpu_cores = getNumberOfPhysicalCPUCores();
+    if (origin_cpu_time_ns <= 0 || total_thread_cnt <= physical_cpu_cores)
+        return toRU(origin_cpu_time_ns);
```

Comment:
why is physical cpu cores

Reply:
Oh, maybe local cpu core is more appropriate